### DISCUSSION TITLE
How can I change the table of contents link to internal link?

### DISCUSSION THREAD

**Discussion by slowdive14** [maximevaillancourt/digital-garden-jekyll-template#105] [General]
As in Footnote, I want to make the link in the table of contents an internal link, but I don't know how. 

[The pages that are struggling with the table of contents are as follows.](https://slowdive.netlify.app/chapter-8-dialectical-dilemmas-and-secondary-targets)

I'm sorry if I missed the same question. 

Please help me.

**Comment by maximevaillancourt** [comment]
Links are converted to "external links" (links that open in a new tab/window) in this function:

https://github.com/maximevaillancourt/digital-garden-jekyll-template/blob/e6e2ea29321aa9ab7e271de943e80ca264ae3dc2/_plugins/open_external_links_in_new_tab.rb#L18-L28

You could ignore links in tables of contents by adding the class of the div that contains the links in the table of contents in this line:

https://github.com/maximevaillancourt/digital-garden-jekyll-template/blob/e6e2ea29321aa9ab7e271de943e80ca264ae3dc2/_plugins/open_external_links_in_new_tab.rb#L23

Essentially, because your links have the `.md-toc-inner` class, you'd want to do this change to prevent them from becoming external links:

```diff
-parsed_doc.css("a:not(.internal-link):not(.footnote):not(.reversefootnote)").each do |link|
+parsed_doc.css("a:not(.internal-link):not(.footnote):not(.reversefootnote):not(.md-toc-inner)").each do |link|
```

**Reply by slowdive14** [reply]
Thank you for your reply. I'll take a more trial-and-error approach based on what you said.
